src/test/java/com/bean00/ArgParserTest.java

```diff
@@ -46,33 +43,80 @@ public void buildServerOptions_returnsTheSecondDirectory_ifMoreThanOneIsPassedIn
         String[] args = {"_", "-d"};
-        ArgParser parser = new ArgParser();
 
-        assertThrows(IllegalArgumentException.class, () -> parser.buildServerOptions(args));
+        assertThrows(IllegalArgumentException.class, () -> ArgParser.buildServerOptions(args));
     }
 
     @Test
     public void buildServerOptions_throwsAnException_ifThereAreNotAnyValidFlags() {
         String[] args = {"_", "_"};
-        ArgParser parser = new ArgParser();
 
-        assertThrows(IllegalArgumentException.class, () -> parser.buildServerOptions(args));
+        assertThrows(IllegalArgumentException.class, () -> ArgParser.buildServerOptions(args));
     }
 
     @Test
     public void buildServerOptions_throwsAnException_ifThereIsOnlyOneArgument_andItIsNotAValidFlag() {
         String[] args = {"_"};
-        ArgParser parser = new ArgParser();
 
-        assertThrows(IllegalArgumentException.class, () -> parser.buildServerOptions(args));
+        assertThrows(IllegalArgumentException.class, () -> ArgParser.buildServerOptions(args));
     }
 
     @Test
     public void buildServerOptions_throwsAnException_ifThereAreZeroArguments() {
         String[] args = {};
-        ArgParser parser = new ArgParser();
 
-        assertThrows(IllegalArgumentException.class, () -> parser.buildServerOptions(args));
+        assertThrows(IllegalArgumentException.class, () -> ArgParser.buildServerOptions(args));
+    }
+
+    @Test
+    public void buildServerOptions_returnsADefaultPort_ifNoPortIsPassedIn() {
+        int expectedPort = 5000;
+        String[] args = {"-d", "path"};
+
+        ServerOptions options = ArgParser.buildServerOptions(args);
+        int port = options.getPort();
+
+        assertEquals(expectedPort, port);
+    }
+
+    @Test
+    public void buildServerOptions_returnsThePort_ifItIsPassedInBeforeTheDirectory() {
+        int expectedPort = 1;
+        String[] args = {"-p", "1", "-d", "path"};
+
+        ServerOptions options = ArgParser.buildServerOptions(args);
+        int port = options.getPort();
+
+        assertEquals(expectedPort, port);
+    }
+
+    @Test
+    public void buildServerOptions_returnsThePort_ifItIsPassedInAfterTheDirectory() {
+        int expectedPort = 1;
+        String[] args = {"-d", "path", "-p", "1"};
+
+        ServerOptions options = ArgParser.buildServerOptions(args);
+        int port = options.getPort();
+
+        assertEquals(expectedPort, port);
+    }
+
+    @Test
+    public void buildServerOptions_returnsTheSecondPort_ifMoreThanOneIsPassedIn() {
+        int expectedPort = 2;
+        String[] args = {"-p", "1", "-p", "2", "-d", "path"};
+
+        ServerOptions options = ArgParser.buildServerOptions(args);
+        int port = options.getPort();
+
+        assertEquals(expectedPort, port);
+    }
+
+    @Test
+    public void buildServerOptions_throwsAnException_ifAnInvalidArgumentIsBetweenThePortArgs() {
+        String[] args = {"-p", "_", "1", "-d", "path"};
```

Comment:
I don't see how this test could pass. You never catch the `NumberFormatException` that I would expect to be thrown in this situation.

Reply:
So for some reason when I ran the test in debug mode, when `Integer.parseInt(portAsString)` was called, an `IllegalArgumentException` was thrown instead of a `NumberFormatException`.

I agree that this is unexpected behavior, and should be accounted for somehow in the code. I'll look into it.

Reply:
I've added a `catch` statement to catch the `NumberFormateException`.